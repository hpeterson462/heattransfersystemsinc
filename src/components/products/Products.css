.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  margin: auto;
  padding: 1%;
  border-radius: 2px;
  background-color:#003366;
}

.card {
  perspective: 1000px;
  margin: 2%;
}

.card:hover .card-inner {
  transform: rotateY(180deg);
}

.card-inner {
  position: relative;
  text-align: center;
  width: 200px;
  height: 100px;
  max-width: 100%;
  transition: transform 0.75s ease-in-out;
  transform-origin: 50% 50%;
  transform-style: preserve-3d;
}

.card-front {
  cursor: pointer;
  color: #003366;
  background-color: #F9FAFB;
  box-shadow: 0 0 5px 2px rgba(0, 0, 0, .1);
}

.product-logo {
  width: 250px;
  height: 100px;
  border-radius: 2px;
}

.card-front, 
  .card-back {
      display: flex;
      position: absolute;
      justify-content: center;
      align-items: center;
      border-radius: 2px;
      width: 100%;
      height: 100%;
      -webkit-backface-visibility: hidden; /* Safari */
      backface-visibility: hidden;
  }

.card-back {
  display: grid;
  font-family: 'Noto Sans', sans-serif;
  transform: rotateY(180deg);
  background-color:#F9FAFB;
}

.ext-link {
  font-size: 25px;
  color: #003366;
}

.ext-link:hover {
  cursor: pointer;
  color: #F68D0B;
}
