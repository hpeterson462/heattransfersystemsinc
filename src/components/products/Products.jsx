import React, { useEffect } from 'react';
import { HiOutlineExternalLink } from "react-icons/hi";
import honeywell from '../../assets/honeywell.png';
import maxonHoneywell from '../../assets/maxonHoneywell.png';
import eclipseHoneywell from '../../assets/eclipseHoneywell.png';
import kromSchroderHoneywell from '../../assets/kromSchroderHoneywell.png';
import hauckHoneywell from '../../assets/hauckHoneywell.png';
import dungs from '../../assets/dungs.png';
import siemens from '../../assets/siemens.png';
import fireye from '../../assets/fireye.png';
import pietroFiorentini from '../../assets/pietroFiorentini.png';
import flynn from '../../assets/flynn.png';
import futureDesignControls from '../../assets/futureDesignControls.png';
import pyromation from '../../assets/pyromation.png';
import exothermicsHoneywell from '../../assets/exothermicsHoneywell.png';
import protectionControlsInc from '../../assets/protectionControlsInc.png';
import asco from '../../assets/asco.png';
import "aos/dist/aos.css";
import AOS from "aos";
import './Products.css';

export default function Products() {

  useEffect(() => {
    AOS.init();
    AOS.refresh();
  }, []);

  const productLogos = [
    {
      name: 'Honeywell',
      image: honeywell,
      url: 'https://thermalsolutions.honeywell.com/us/en/home',
      description: 'Trusted technology to manage your burners.'
    },
    {
      name: 'Honeywell Maxon',
      image: maxonHoneywell,
      url: 'https://thermalsolutions.honeywell.com/us/en/home',
      description: 'Trusted technology to manage your burners.'
    },
    {
      name: 'Honeywell Eclipse',
      image: eclipseHoneywell,
      url: 'https://thermalsolutions.honeywell.com/us/en/home',
      description: 'Trusted technology to manage your burners.'
    },
    {
      name: 'Honeywell Krom Schroder',
      image: kromSchroderHoneywell,
      url: 'https://www.kromschroeder.de/en/',
      description: 'We control your furnaces & your success.'
    },
    {
      name: 'Honeywell Hauck',
      image: hauckHoneywell,
      url: 'https://thermalsolutions.honeywell.com/us/en/industries/hauck',
      description: 'Asphalt & aggregate drying.'
    },
    {
      name: 'Karl Dungs Inc.',
      image: dungs,
      url: 'https://www.dungs.com/en',
      description: 'Safe & clean gas combustion.'
    },
    {
      name: 'Siemens',
      image: siemens,
      url: 'https://scccombustion.com/',
      description: 'Siemens Combustion Controls Inc.'
    },
    {
      name: 'Fireye',
      image: fireye,
      url: 'https://www.fireye.com/',
      description: 'Flame safeguard controls & burner systems.'
    },
    {
      name: 'Pietro Fiorentini',
      image: pietroFiorentini,
      url: 'https://www.fiorentini.com/us/en',
      description: 'Solutions for oil & gas streams.'
    },
    {
      name: 'Flynn',
      image: flynn,
      url: 'http://flynnburner.com/',
      description: 'Combustion components for the bakery industry.'
    },
    {
      name: 'Future Design Controls',
      image: futureDesignControls,
      url: 'http://www.futuredesigncontrols.com/',
      description: 'Recording & control products.'
    },
    {
      name: 'Pyromation',
      image: pyromation,
      url: 'https://www.pyromation.com/',
      description: 'Temperature sensors & assemblies.'
    },
    {
      name: 'Honeywell Exothermics',
      image: exothermicsHoneywell,
      url: 'https://thermalsolutions.honeywell.com/us/en/home',
      description: 'Trusted technology to manage your burners.'
    },
    {
      name: 'Protection Controls Inc.',
      image: protectionControlsInc,
      url: 'https://www.protectioncontrolsinc.com/',
      description: 'Electrical flame safety equipment.'
    },
    {
      name: 'Asco',
      image: asco,
      url: 'https://www.emerson.com/en-us/automation/asco',
      description: 'Fluid automation solutions.'
    },
  ]

  const displayProductLogos = productLogos.map(
    (productLogos, index) =>
      <div className="card" key={productLogos.name + index}>
        <div className="card-inner">
          <div className="card-front">
            <img
              className="product-logo"
              src={productLogos.image}
              alt={`${productLogos.name} `}
            />
          </div>
          <div className="card-back">
            <p>{productLogos.description}</p>
            <div>
              <a
                className="ext-link"
                target="_blank"
                href={productLogos.url}
              >
                <HiOutlineExternalLink />
              </a>
            </div>
            <div />
          </div>
        </div>
      </div >
  )

  return (
    <section>
      <h1 id="products">
        Products
      </h1>

      <h2>
        We offer a variety of quality products from these companies.
      </h2>

      <div className="card-container" data-aos="fade-zoom-in" data-aos-offset="200" data-aos-easing="ease-in-sine" data-aos-duration="900">

        {displayProductLogos}

      </div>
    </section>
  )
}
