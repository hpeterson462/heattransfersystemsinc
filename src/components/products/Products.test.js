import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import Products from './Products';

describe('Products component', () => {
  afterEach(() => cleanup());

  it('renders Products component', () => {
    const { asFragment } = render(<MemoryRouter><Products /></MemoryRouter>);

    expect(asFragment()).toMatchSnapshot();
  });
});
