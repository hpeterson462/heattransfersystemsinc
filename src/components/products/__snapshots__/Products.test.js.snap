// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products component renders Products component 1`] = `
<DocumentFragment>
  <section>
    <h1
      id="products"
    >
      Products
    </h1>
    <h2>
      We offer a variety of quality products from these companies.
    </h2>
    <div
      class="card-container"
    >
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Honeywell "
              class="product-logo"
              src="honeywell.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Trusted technology to manage your burners.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://thermalsolutions.honeywell.com/us/en/home"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Honeywell Maxon "
              class="product-logo"
              src="maxonHoneywell.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Trusted technology to manage your burners.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://thermalsolutions.honeywell.com/us/en/home"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Honeywell Eclipse "
              class="product-logo"
              src="eclipseHoneywell.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Trusted technology to manage your burners.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://thermalsolutions.honeywell.com/us/en/home"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Honeywell Krom Schroder "
              class="product-logo"
              src="kromSchroderHoneywell.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              We control your furnaces & your success.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://www.kromschroeder.de/en/"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Honeywell Hauck "
              class="product-logo"
              src="hauckHoneywell.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Asphalt & aggregate drying.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://thermalsolutions.honeywell.com/us/en/industries/hauck"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Karl Dungs Inc. "
              class="product-logo"
              src="dungs.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Safe & clean gas combustion.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://www.dungs.com/en"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Siemens "
              class="product-logo"
              src="siemens.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              SSC Inc.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://scccombustion.com/"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Fireye "
              class="product-logo"
              src="fireye.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Flame safeguard controls & burner systems.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://www.fireye.com/"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Pietro Fiorentini "
              class="product-logo"
              src="pietroFiorentini.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Solutions for oil & gas streams.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://www.fiorentini.com/us/en"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Flynn "
              class="product-logo"
              src="flynn.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Combustion components for the bakery industry.
            </p>
            <div>
              <a
                class="ext-link"
                href="http://flynnburner.com/"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Future Design Controls "
              class="product-logo"
              src="futureDesignControls.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Recording & control products.
            </p>
            <div>
              <a
                class="ext-link"
                href="http://www.futuredesigncontrols.com/"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Pyromation "
              class="product-logo"
              src="pyromation.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Temperature sensors & assemblies.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://www.pyromation.com/"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Honeywell Exothermics "
              class="product-logo"
              src="exothermicsHoneywell.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Trusted technology to manage your burners.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://thermalsolutions.honeywell.com/us/en/home"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Protection Controls Inc. "
              class="product-logo"
              src="protectionControlsInc.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Electrical flame safety equipment.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://www.protectioncontrolsinc.com/"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-inner"
        >
          <div
            class="card-front"
          >
            <img
              alt="Asco "
              class="product-logo"
              src="asco.png"
            />
          </div>
          <div
            class="card-back"
          >
            <p>
              Fluid automation solutions.
            </p>
            <div>
              <a
                class="ext-link"
                href="https://www.emerson.com/en-us/automation/asco"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
